### DISCUSSION TITLE
Is there any reason why GetCandidateSpecificity does not include `U*INT` data types?

### DISCUSSION THREAD

**Discussion by pkit** [duckdb/duckdb#10648] [Q&A]
I cannot see any comment on why the current subset was used in the code.
`UINTEGER` is more specific than `INTEGER` and pretty easily detectable. 
What about `UUID`? It's more specific than `VARCHAR` and can be deduced too.
And so on.
I'm talking about here, just in case:
https://github.com/duckdb/duckdb/blob/20b1486d1192f9fbd2328d1122b5afe5f1747fce/src/execution/operator/csv_scanner/util/csv_reader_options.cpp#L351-L360

**Comment by Tishj** [comment]
This is auto detection, every added type likely incurs a substantial cost, that does not mean you cant override the type when invoking read_csv

**Reply by pkit** [reply]
I'm not sure what "override the type" means here.
Afair, auto detection is handled by a subset of these types from the `auto_types_candidates`.

**Reply by Tishj** [reply]
Yes, that's auto detection, you can use `types` to set them explicitly for a given column

**Reply by pkit** [reply]
I don't need explicitly. I need auto_detection for uints. I have PR ready and will benchmark it.
But even theoretically I don't see why it should be slower. The real auto-detection type set is always: null, bool, bigint, float, time, datetime, varchar. IIRC.
How that will change if more types are added to `GetCandidateSpecificity`?

**Reply by pkit** [reply]
In the end my question is: was there some special consideration about these types or it is there only because the list of types was hardcoded earlier? (before the `auto_types_candidates` was inroduced)
